sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -665,15 +665,7 @@ case class AlterTableSetLocationCommand(
         catalog.alterPartitions(tableName, Seq(newPart))
       case None =>
         // No partition spec is specified, so we set the location for the table itself
-        val newTable =
-          if (DDLUtils.isDatasourceTable(table)) {
-            table.withNewStorage(
-              locationUri = Some(location),
-              properties = table.storage.properties ++ Map("path" -> location))
-          } else {
-            table.withNewStorage(locationUri = Some(location))
-          }
-        catalog.alterTable(newTable)
+        catalog.alterTable(table.withNewStorage(locationUri = Some(location)))
```

Comment:
FYI we have a bug here, currently we allow users to `SET LOCATION` for managed data source table, however, in `SHOW CREATE TABLE`, we can't generate corrected SQL to create managed data source table whose location has been set, because data source with path is always treated as external table. We should either forbid `SET LOCATION` for managed data source table, or improve the CREATE TABLE syntax to support managed table with path.